lib/index.js

```diff
@@ -337,6 +343,42 @@ Server.prototype = {
       yield this.apiml.registerMainServerInstance();
     }
   }),
+  
+  configureApimlStorage(apimlConfig) {
+    const { gatewayHost, gatewayPort } = apimlConfig.server;
+    const { cachingService } = apimlConfig;
+    if (typeof gatewayHost !== 'string' || typeof gatewayPort !== 'number') {
+      return;
+    }
+    if (typeof cachingService !== 'object') {
+      return;
+    }
+    const {cert: clientCertFile, key: clientKeyFile} = cachingService;
+    if (!clientCertFile || !clientKeyFile) {
+      return;
+    }
+    let cert, key;
+    try {
+      cert = fs.readFileSync(clientCertFile);
```

Comment:
Is it guaranteed to be a file? Is it guaranteed to be a base64 PEM file? Because p12, DER encoding, and keyrings are all possible alternatives for https keys in zowe and I think this code should really be using the logic in webserver.js in order to handle different types.

Reply:
Currently, `node.https` are not good to use with Caching Service so we need a way to provide a valid cert/key pair.
I a little bit changed the code: now tlsOptions from `node.https` are used for Caching Service unless they are overridden by `mediationLayer.cachingService.https`. 
Reusing `readTlsOptionsFromConfig` from webserver.js is a good idea. Thanks!